nvtabular/ops.py

```diff
@@ -1230,3 +1230,48 @@ def def_emb_sz(self, classes, n, sz_dict=None):
         sz = sz_dict.get(n, int(self.emb_sz_rule(n_cat)))  # rule of thumb
         self.embed_sz[n] = sz
         return n_cat, sz
+
+
+class DFLambda(DFOperator):
+    """
+    Enables to call Methods to cudf.Series
+
+    Parameters
+    -----------
+    f : lambda function
+        defines the function executed on dataframe level
+    name_new : str
+        name of the output column, if column should be replaced it needs to have the same name
+    columns :
+    preprocessing : bool, default True
+        Sets if this is a pre-processing operation or not
+    replace : bool, default True
+        Replaces the transformed column with the original input
+        if set Yes
+    """
+
+    default_in = ALL
+    default_out = ALL
+
+    def __init__(self, f=None, name_new=None, columns=None, preprocessing=True, replace=True):
+        super().__init__(columns=columns, preprocessing=preprocessing, replace=replace)
+        self.f = f
+        self.name_new = name_new
+        self.replace = replace
+        self.preprocessing = preprocessing
+
+    @property
+    def req_stats(self):
+        return []
+
+    @annotate("DFLambda_op", color="darkgreen", domain="nvt_python")
+    def op_logic(self, gdf: cudf.DataFrame, target_columns: list, stats_context=None):
+        if self.replace and self.preprocessing:
+            new_gdf = gdf
+        else:
+            new_gdf = cudf.DataFrame()
+
+        new_gdf[self.name_new] = self.f(gdf)
+        if self.replace and self.preprocessing:
+            new_gdf = new_gdf[target_columns]
+        return new_gdf
```

Comment:
I tried it out and I had add line 1275 and 1276 to order the columns in the same order as target_columns. If I did not add these lines, it will overwrites the wrong column.

Example below, If I did not add line 1275 and 1276, the output would be written in column 'num'

assemble_new_df assumes the columns are in the same order, but it does not check it.
`origin_gdf[target_columns] = new_gdf`

```
    def assemble_new_df(self, origin_gdf, new_gdf, target_columns):
        if self.replace and self.preprocessing and target_columns:
            if new_gdf.shape[0] < origin_gdf.shape[0]:
                return new_gdf
            else:
                origin_gdf[target_columns] = new_gdf
                return origin_gdf
        return cudf.concat([origin_gdf, new_gdf], axis=1)
```

```
cat_1 = np.asarray(['aaaaa']*12 + ['bvvvv']*10 + ['ddddc']*10)
num_1 = np.asarray([1,1,2,2,2,1,1,5,4,4,4,4, 1,2,3,4,5,6,7,8,9,10, 1,2,3,4,5,6,7,8,9,10])
pdf_1 = pd.DataFrame({'cat': cat_1, 'num': num_1})

pdf_1.to_parquet('pdf_1.parquet')

proc = nvt.Workflow(
    cat_names=['cat'],
    cont_names=['num'],
    label_name=[],
    to_cpu=False)

proc.add_preprocess(
    DFLambda(
        f=lambda gdf: gdf['cat'].str.slice(1,3), 
        name_new='cat',
        preprocessing=True,
        replace=True
    )
)

input_files = glob.glob('./pdf_1.parquet')

train_ds_iterator = nvt.dataset(input_files, engine='parquet')

proc.apply(train_ds_iterator, 
           apply_offline=True, 
           record_stats=True, 
           shuffle=False, 
           output_path='./')
```

Reply:
I think it should work now